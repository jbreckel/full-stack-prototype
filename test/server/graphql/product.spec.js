import gql from 'graphql-tag'

import execute from './mockServer'

describe('product', () => {
  it('can query products', (done) => {
    execute(gql`
      query {
        products {
          id
          name
          purchasePrice
          salePrice
          category {
            id
            name
          }
        }
      }
    `)
      .then((result) => {
        expect(result).toMatchSnapshot()
        done()
      })
  })

  it('can query product with id', (done) => {
    execute(gql`
      query getProduct($id: ID!){
        product(id: $id) {
          id
          name
          purchasePrice
          salePrice
          category {
            id
            name
          }
        }
      }
    `,
      {
        id: 'id',
      })
      .then((result) => {
        expect(result).toMatchSnapshot()
        done()
      })
  })
})
