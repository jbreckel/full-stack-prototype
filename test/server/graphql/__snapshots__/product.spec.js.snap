// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product can query product with id 1`] = `
Object {
  "data": Object {
    "product": Object {
      "category": Object {
        "id": "id",
        "name": "name",
      },
      "id": "id",
      "name": "name",
      "purchasePrice": 1.5,
      "salePrice": 1.5,
    },
  },
}
`;

exports[`product can query products 1`] = `
Object {
  "data": Object {
    "products": Array [
      Object {
        "category": Object {
          "id": "id",
          "name": "name",
        },
        "id": "id",
        "name": "name",
        "purchasePrice": 1.5,
        "salePrice": 1.5,
      },
      Object {
        "category": Object {
          "id": "id",
          "name": "name",
        },
        "id": "id",
        "name": "name",
        "purchasePrice": 1.5,
        "salePrice": 1.5,
      },
    ],
  },
}
`;
